Assets/Scripts/RedRunner/Characters/RedCharacter.cs

```diff
@@ -263,21 +270,39 @@ public override AudioSource Audio
 
 		void Awake ()
 		{
-			m_InitialPosition = transform.position;
-			m_InitialScale = transform.localScale;
-			m_GroundCheck.OnGrounded += GroundCheck_OnGrounded;
-			m_Skeleton.OnActiveChanged += Skeleton_OnActiveChanged;
+            m_InitialPosition = transform.position;
+            m_InitialScale = transform.localScale;
+            m_GroundCheck.OnGrounded += GroundCheck_OnGrounded;
+            m_Skeleton.OnActiveChanged += Skeleton_OnActiveChanged;
             IsDead = new Property<bool>(false);
-			m_ClosingEye = false;
-			m_Guard = false;
-			m_Block = false;
-			m_CurrentFootstepSoundIndex = 0;
-			GameManager.OnReset += GameManager_OnReset;
+            m_ClosingEye = false;
+            m_Guard = false;
+            m_Block = false;
+            m_CurrentFootstepSoundIndex = 0;
+            GameManager.OnReset += GameManager_OnReset;
+
+            // Default to static rigidbodies.
+            // We don't want to perform physics simulations for other players.
+            m_Rigidbody2D.bodyType = RigidbodyType2D.Static;
+
+            LocalPlayerSpawned += () =>
+            {
+                // Once we find out we are the local player, simulate our rigidbody.
+                Local.m_Rigidbody2D.bodyType = RigidbodyType2D.Dynamic;
+            };
 		}
+        
+        public override void OnStartLocalPlayer()
+        {
+            base.OnStartLocalPlayer();
 
-		void Update ()
+            Local = this;
+            LocalPlayerSpawned();
+        }
+
+        void Update ()
 		{
-			if ( !GameManager.Singleton.gameStarted || !GameManager.Singleton.gameRunning )
```

Comment:
Why did you get rid of this check?

Reply:
The player no longer exists in the scene when the game starts. We only spawn the player once you join the network after hitting play, so both these checks will always be true.